### PR TITLE
added the drawDominoes game mode to Multiplayer Dominoes

### PR DESCRIPTION
**Pull request by ilikecheese117**
Implemented the drawDominoes game mode in Multiplayer Dominoes.

- Created button to start the game mode from the /gamemode page.
- Added game mode rules to the rule engine. These include rules to count score for the winning player, validate domino moves and to end the game once a player has reached at least 100 points.

Game mode based on a 1v1 version of the draw game described here (this means no boneyard for now) https://www.dominorules.com/draw

### CODE REVIEW

**Comment by fernandopizarro1** [comment]
> Implemented the draw Dominoes game mode in the rule engine

Please provide a better description!

**Review by fernandopizarro1 — CHANGES_REQUESTED** [review]
You should implement the ruleset for the drawDominoes instead off just the placeholder.

**Comment by ilikecheese117** [comment]
> You should implement the ruleset for the drawDominoes instead off just the placeholder.

Added changes! @fernandopizarro1

**Review by AngelCIICMorales — CHANGES_REQUESTED** [review]
- Merge the current main so the tile placements get fixed so I can properly do some tests.
- The bot ends in an infinite loop of placing tiles.
- There are more than 28 tiles.
image

**Comment by ilikecheese117** [comment]
> * Merge the current main so the tile placements get fixed so I can properly do some tests.
> * The bot ends in an infinite loop of placing tiles.
> * There are more than 28 tiles.
>   image

Merged to main. This seems to be an animation glitch rather than a draw dominoes glitch? Draw dominoes right now just uses the validate move function which is the exact same for all rule sets (including draw dominoes). Additionally added scoring to draw dominoes implementation.

**Comment by ilikecheese117** [comment]
> You should implement the ruleset for the drawDominoes instead off just the placeholder.

@fernandopizarro1 merged to main and added proper scoring to the game mode

File: frontend/src/gamestate/experimentalGame.js [hunk 1]
```diff
@@ -17,26 +17,49 @@ const tileMap = new Map();
...
-        const handleCanPlay = () => {
-          audio.play().catch((error) => {
-            console.log("Autoplay was prevented, retrying with mute...", error);
-            audio.muted = true; // Mute the audio if autoplay is blocked
-            audio.play();
-          });
+    const [playerScore, setPlayerScore] = useState(0);
+    const [botScore, setBotScore] = useState(0);
+
+    const BackgroundMusic = ({ src }) => {
```
Comment (fernandopizarro1) [inline frontend/src/gamestate/experimentalGame.js:23]: Consider moving the audio into a UseEffect, with an empty array as a dependency, so there is always one Audio component created.

Comment (fernandopizarro1) [reply frontend/src/gamestate/experimentalGame.js:23]: Try this:
```javascript
const audioRef = useRef(null);
const [isPlaying, setIsPlaying] = useState(false);
const [volume, setVolume] = useState(0.5);

useEffect(() => {
  audioRef.current = new Audio(src);
  audioRef.current.volume = volume;

  return () => {
    audioRef.current.pause();
    audioRef.current = null; 
  };
}, [src]);
```

Comment (ilikecheese117) [reply frontend/src/gamestate/experimentalGame.js:23]: Already tried this.

**Review by fernandopizarro1 — APPROVED** [review]

**Review by AngelCIICMorales — APPROVED** [review]
LGTM

**Pull request merged**